cg/models/demultiplex/run_parameters.py

```diff
@@ -101,15 +99,44 @@ def get_read_2_cycles(self) -> int | None:
         """Return the number of cycles in the second read."""
         pass
 
+    @property
+    def is_single_index(self) -> bool:
+        """Returns true if the sequencing run is single-index."""
+        return self.get_index_2_cycles() == 0
+
+    def _is_novaseq6000_post_1_5_kit(self) -> bool:
+        """
+        Returns whether sequencing was performed after the 1.5 consumables kits where introduced.
+        This is indicated by the software version and the reagent kit fields in the run parameters.
+        """
+        if self.sequencer != Sequencers.NOVASEQ:
+            return False
+        if parse(self.control_software_version) < parse(NEW_NOVASEQ_CONTROL_SOFTWARE_VERSION):
+            return False
+        if parse(self.reagent_kit_version) < parse(NEW_NOVASEQ_REAGENT_KIT_VERSION):
+            return False
+        return True
+
+    @property
+    def index_settings(self) -> IndexSettings:
+        """Returns the correct index-related settings for the run in question."""
+        if self.sequencer == Sequencers.NOVASEQX:
+            LOG.debug("Using NovaSeqX index settings")
+            return NOVASEQ_X_INDEX_SETTINGS
+        if self._is_novaseq6000_post_1_5_kit():
+            LOG.debug("Using NovaSeq 6000 post 1.5 kits index settings")
+            return NOVASEQ_6000_POST_1_5_KITS
+        return NO_REVERSE_COMPLEMENTS
```

Comment:
I would prefer to not have this as a property, but an attribute that is set once. As a property the code will be run every time it is called, and since the code parses xml files it seems quite unnecessary. Since the value will not change after instantiating the model I think it is ok to just set it once

Reply:
fixed